test/tests/pbs_accumulate_resc_used.py

```diff
@@ -0,0 +1,1456 @@
+             'job_state': 'F'}
+        self.server.expect(JOB, a, extend='x',
+                           offset=30, interval=1,
+                           max_attempts=20, id=jid)
+
+        # Restart server and verifies that the values are still the same
+        self.server.restart()
+
+        # Below is commented due to a known PBS issue
+        # self.server.expect(JOB, a, extend='x', id=jid)
+
+    def test_server_restart(self):
+        """
+        Test that resource accumulation will not get
+        impacted if sever is restarted during job execution
+        """
+
+        # Create a prologue hook
+        hook_body = """
+import pbs
+e=pbs.event()
+pbs.logmsg(pbs.LOG_DEBUG, "executed prologue hook")
+if e.job.in_ms_mom():
+    e.job.resources_used["vmem"] = pbs.size("11gb")
+    e.job.resources_used["foo_i"] = 11
+    e.job.resources_used["foo_f"] = 0.11
+    e.job.resources_used["foo_str"] = '{"seven":7}'
+    e.job.resources_used["cput"] = 11
+    e.job.resources_used["stra"] = '"glad,elated","happy"'
+    e.job.resources_used["foo_str4"] = "eight"
+else:
+    e.job.resources_used["vmem"] = pbs.size("12gb")
+    e.job.resources_used["foo_i"] = 12
+    e.job.resources_used["foo_f"] = 0.12
+    e.job.resources_used["foo_str"] = '{"eight":8,"nine":9}'
+    e.job.resources_used["cput"] = 12
+    e.job.resources_used["stra"] = '"cucumbers,bananas"'
+"""
+
+        hook_name = "prolo"
+        a = {'event': "execjob_prologue", 'enabled': 'True'}
+        self.server.create_import_hook(
+            hook_name,
+            a,
+            hook_body,
+            overwrite=True)
+
+        a = {'Resource_List.select': '3:ncpus=1',
+             'Resource_List.walltime': 10,
+             'Resource_List.place': 'scatter'}
+        j = Job(TEST_USER)
+        j.set_attributes(a)
+        j.set_sleep_time("5")
+
+        # The pbsdsh call is what allows a first task to get spawned on
+        # on a sister mom, causing the execjob_prologue hook to execute.
+        j.create_script(
+            "pbsdsh -n 1 hostname\n" +
+            "pbsdsh -n 2 hostname\n" +
+            "sleep 300\n")
+
+        jid = self.server.submit(j)
+
+        # Once the job is started running restart server
+        self.server.expect(JOB, {'job_state': "R", "substate": 42}, id=jid)
+        self.server.restart()
+
+        # Job will be requeued and rerun. Verify that the
+        # resource accumulation is similar as if server is
+        # not started
+        a = {'resources_used.foo_i': '35',
+             'resources_used.foo_f': '0.35',
+             'resources_used.vmem': '35gb',
+             'resources_used.cput': '00:00:35',
+             'resources_used.stra': "\"glad,elated\",\"happy\"",
+             'resources_used.foo_str4': "eight",
+             'job_state': 'F'}
+        self.server.expect(JOB, a, extend='x',
+                           offset=5, id=jid, interval=1)
+
+        foo_str_dict_in = eval('{"eight": 8, "seven": 7, "nine": 9}')
```

Comment:
Same as the previous comment about directly setting dictionaries vs using eval()